lib/pharos/salt.rb

```diff
@@ -16,51 +17,102 @@ def initialize
     # Returns two values:
     #   - The response object.
     #   - A hash containing the parsed JSON response.
-    def call(tgt, fun, arg = nil)
-      login! unless @token
-
+    def call(tgt:, fun:, arg: nil)
       hsh = { tgt: tgt, fun: fun, client: "local" }
       hsh[:arg] = arg if arg
 
-      res = perform_request("/", "post", hsh)
+      res = perform_request(endpoint: "/", method: "post", data: hsh)
       [res, JSON.parse(res.body)]
     end
 
     # Returns the minions as discovered by salt.
-    def minions(mid = nil)
-      login! unless @token
-
-      id  = mid ? "/#{mid}" : ""
-      res = perform_request("/minions#{id}", "get")
+    def minions
+      res = perform_request(endpoint: "/minions", method: "get")
       JSON.parse(res.body)["return"].first
     end
 
+    # Return information for the minion linked to this instance.
+    def minion
```

Comment:
The methods of this class behave differently when a minion id is assigned and this can lead to confusion. For example if we see a line of code like: ```salt.minion``` somewhere in the code we don't know if it returns nil because a minion was not provided (bug?) or because the minion doesn't exist. Also the request will be sent even if no minion id is specified which will always return nil (because of the last line in this method). 

I think this library should not even be instantiated but rather be used as a module providing a number of methods directly mapped to API requests. Then the Minion model can implement all the methods related to that minion that will call the methods of this class with the right arguments.